dask_expr/tests/test_collection.py

```diff
@@ -1197,3 +1197,18 @@ def test_filter_pushdown(df, pdf):
     df_opt = df[["x", "y"]].simplify().replace(1, 5)
     expected = df_opt[df_opt.x > 5]
     assert result._name == expected._name
+
+
+def test_shape(df, pdf):
+    result = df.shape
+    assert result[0]._name == (df.size / 2)._name
```

Comment:
Do we really want to test that the implementation of `shape` is related to `size` in this way. If they both return a correct result when computed, then do we really "care" if this assertion fails?

Also can we use something like `pdf.shape[1]` instead of `2`? (or set `expect = pdf.shape` above, etc)

Reply:
That's what we are doing in dask/dask as well, so I don't think that this hurts? I want to validate that the row-dimension is not eager, we could alternatively check for a Expression or so

Reply:
>I don't think that this hurts?

No, it doesn't hurt. I'm just hoping to avoid white-box testing when we don't really need it, since I noticed such tests have led to tech debt in dask/dask. This case is not particularly problematic either way, so I'll leave it up to you.

Feel free to merge if/when you think this is ready :)